### PR TITLE
Proper human friendly URLs for kiwix-serve

### LINKED ISSUE #676: HTTP daemon does not have a proper human friendly name if not started with kiwix-serve
If started with Kiwix Desktop, URL look like:
http://192.168.0.101:8181/search?content=5149203d-1c60-a927-0095-278e34e6897a&pattern=estrutura+biologica

I suspect the problem comes from the `humanfriendly()` function which probably falls back.

This function should be fixed and in place of using the ZIM filename (which probably is not available here), it should build the human friendly name from the ZIM metadata (name, publisher, date, flavour).

### PR DESCRIPTION
**Pull request by MananJethwani**
this resolves #676

### CODE REVIEW

**Review by kelson42 — APPROVED** [review]
From a user perspective, seems to work fine

File: src/kiwixapp.h [hunk 1]
```diff
@@ -16,6 +16,8 @@
 #include <QApplication>
 #include <QErrorMessage>
 #include <QTranslator>
+#include <kiwix/name_mapper.h>
+#include <kiwix/tools.h>
```
Comment (mgautierfr) [inline src/kiwixapp.h:20]: Why do you need to include tools ?

Comment (kelson42) [reply src/kiwixapp.h:20]: Seems useless indeed. Compile and run properly after my rebase and removal.

**Comment by kelson42** [comment]
@MananJethwani @mgautierfr Can we merge this?

**Comment by MananJethwani** [comment]
@kelson42 I believe we can merge this waiting for @mgautierfr.

**Review by mgautierfr — APPROVED** [review]

**Pull request merged**